src/main/java/org/ambraproject/wombat/controller/ArticleController.java

```diff
@@ -834,9 +834,13 @@ private XmlContent getXmlContent(Site site, ArticlePointer articlePointer,
         Reference ref = itRef.next();
         if (ref.getDoi() != null && ref.getDoi().startsWith("10.1371/")) {
           String key = itKey.next();
-          Reference.Builder builder = new Reference.Builder(ref);
-          Reference refWithLink = builder.setFullArticleLink(getLinkText(site, request, ref.getDoi(), key)).build();
-          referencesWithLinks.add(refWithLink);
+          if (!Strings.isNullOrEmpty(key)) {
```

Comment:
nit: I usually avoid the **not:** **not** logic.  You can rewrite the `if`:
```
if (Strings.isNullOrEmpty(key)) {
    referencesWithLinks.add(ref);
} else {
    Reference.Builder builder = new Reference.Builder(ref);
    Reference refWithLink = builder.setFullArticleLink(getLinkText(site, request, ref.getDoi(), key)).build();
    referencesWithLinks.add(refWithLink);
}
```
Just a suggestion :)